include/exadg/structure/spatial_discretization/operator.cpp

```diff
@@ -740,7 +736,13 @@ Operator<dim, Number>::solve_nonlinear(VectorType &       sol,
   // solve nonlinear problem
   auto const iter = newton_solver->solve(sol, update);
 
-  // set inhomogeneous Dirichlet values
+  // This step should actually be optional: The constraints have already been set
+  // before the nonlinear solver is called and no contributions to the constrained
+  // degrees of freedom should be added in the Newton solver by the linearized solver
+  // (because the residual vector forming the rhs of the linearized problem is zero
+  // for constrained degrees of freedom, the initial solution of the linearized
+  // solver is also zero, and the linearized operator contains values of 1 on the
+  // diagonal for constrained degrees of freedom.
```

Comment:
Tiny comment: I think you need to close the braces
```suggestion
  // diagonal for constrained degrees of freedom).
```